.github/pull_request_template.md

```diff
@@ -0,0 +1,24 @@
+Thank you for taking the time to contribute to this proposal!
+We’re so happy you’re helping out.
+
+1. Please take a look at the [contributing guidelines][]
+   and the resources to which it links.
+2. Please include the purpose of the pull request. For example:
+   * “This is adds…”
+   * “This simplifies…”
+   * “This fixes…”
```

Comment:
I would be inclined in adding some Issue Templates that forces people to think a bit about why they are creating an issue, same with Pull Request Templates.

By now, the TC39 has done an amazing job closing a lot of issues (I subscribed to all and I read them all), but people keep opening issues to rant about things and start their own subthread about the exact same topic.

I would appreciate it if you could help on that front.

Reply:
Thanks for the comment. I agree and I plan to add issue templates after I merge this pull request. (GitHub now has a special graphical interface for creating issue templates, and its old text-based system is apparently legacy now. Otherwise I would have created the templates in this pull request

I would love to get perspectives on what issue templates would be good to add and what sort of things should be in them.

Reply:
I am not sure exactly since I honestly don't understand the TC39 strategy, so I will be asking few questions.

Do we want to send people to the same thread to talk about the following topics?:

A. Capture Operator
B. F#

---- 

My proposal, F# goes to one thread where people keep sharing their ideas.

```yaml
blank_issues_enabled: false # super important this part
contact_links:
  - name: Discussion about F# Pipe
    url:  https://github.com/tc39/proposal-pipeline-operator/discussions/categories/f-sharp
    about: Please share your thoughts about F# Pipe
  - name: Discussion about Placeholder
    url:  https://github.com/tc39/proposal-pipeline-operator/discussions/categories/placeholder
    about: Please share your thoughts about the placeholder
  - name: Help or Question❓
    url: https://github.com/tc39/proposal-pipeline-operator/discussions/categories/q-a
    about: Please ask and answer questions here.
```

Worth noting, the link is pointing to `Discussions` rather than Issues, I will strongly suggest enabling Discussions with the #default config, and add such category as I shared before.

----

The previous list will filter a lot of the noise at the moment, and if we do that.

What would mean opening an issue?

My proposal, TC39 could start leveraging Issues as something they must take ownership in closing, and probably help themselves to be a Ticket system for potential tasks, amendments, and stuff like that.

----

#### Important thought about an Issue

Has been quote a disrespectful and cult-link behavior, adding a field to the issue that acknowledges that they are not opening an issue related to F# (since they suppose to use the Discussions) will be something that I personally would appreciate.

The point is not to avoid having conversations, but rather, don't abuse the Issues where people like me are on top of `all the updates` that actually affect the repository moving forward.

----

Is that a reasonable proposal?

Reply:
> I am not sure exactly since I honestly don't understand the TC39 strategy, so I will be asking few questions.
> 
> Do we want to send people to the same thread to talk about the following topics?:
> 
> A. Capture Operator
> B. F#

This is a good fundamental question, and it is one to which I don’t yet have a definite answer. I need to talk with other champions first before deciding on what our community goals are here.

> My proposal, F# goes to one thread where people keep sharing their ideas.
> 
> Worth noting, the link is pointing to `Discussions` rather than Issues, I will strongly suggest enabling Discussions with the #default config, and add such category as I shared before.
> 
> The previous list will filter a lot of the noise at the moment, and if we do that.

I am personally enthusiastic about GitHub Discussions, and I actually have asked about enabling them for this proposal (it would be the first TC39 proposal to enable them). However, when I asked the others about enabling them, [TC39 representatives both inside this proposal and outside this proposal gave negative responses](https://view.matrix.org/room/!WgJwmjBNZEXhJnXHXw:matrix.org/?anchor=$CZ8U36b5xFYVFKJ0bi4yamkDFFz5wbyjy7ymKy_DvJY&offset=120). They generally don’t want another place to check. (See also #209.)

> My proposal, TC39 could start leveraging Issues as something they must take ownership in closing, and probably help themselves to be a Ticket system for potential tasks, amendments, and stuff like that.

I agree with this. However, we champions will need to make clear criteria for when it would be appropriate to close an issue (when an issue is “finished”). As before, I plan to talk with other champions first before deciding on what our community goals are here.

> Has been really a disrespectful and cult-link behavior around F# topic, adding a field to the issue that acknowledges that they are not opening an issue related to F# (since they suppose to use the Discussions) will be something that I personally would appreciate.

I do want to (gently) ask that we not refer to any behavior as “cult-like”, however much we feel that it is. Such [strong language][cult] probably doesn’t fulfill our [code of conduct][]’s principles of being “friendly and patient” and “careful in the words we choose”. (The word “disrespectful” is fine, of course, although I would be careful even with that word.) We want to try our best not to alienate anyone here.

[cult]: https://en.wikipedia.org/wiki/Cult
[code of conduct]: https://tc39.es/code-of-conduct/

> The point is not to avoid having conversations, but rather, don't abuse the Issues where people like me are on top of `all the updates` that actually affect the repository moving forward.

Yes, this is a reasonable goal. I will need to talk about it with the other champions more.

We don’t want to shut down conversations. But we are all volunteers, and we simply are not able to manage and moderate very high volumes of discussion every day. We’ll figure this out after I merge this pull request’s basic community-health files.

Reply:
> I do want to (gently) ask that we not refer to anyone’s behavior as “cult-like”,

Understood,  just frustrated to read titles like this one https://github.com/tc39/proposal-pipeline-operator/issues/205 and be so passive about it. 

(Beside the word chosen, I understand that) I think we can't afford to have such a mindset around decisions that will affect all the incoming generations `forever`.

Reply:
I don't want to enable Discussions just as a way of firewalling threads we don't want to read. It's disrespectful to the time people put into things.

If we don't want to receive additional issues about F#-style pipes, we can possibly put that into the issue template, and be proactive in closing new threads that are largely just about the F#-style syntax.

Reply:
I would think of Discussions more as a way of separating “concrete actionable bugs in the specification and explainer” (Issues) from “longer-term discussion about broader topics”, like follow-on proposals or fundamental approaches. But I also understand the fact that lots of people don’t want yet another place to check for important discussion.

Having said that, I think it’ll take a while to figure out how best to moderate topics about F# pipes (and what our goals with F#-pipe issues should be—continued discussion? avoiding going in circles? strengthening the explainer? creating follow-up proposals?). So writing issue templates will take longer to do.